QModManager/API/SMLHelper/Assets/Spawnable.cs

```diff
@@ -0,0 +1,128 @@
+        /// </summary>
+        public bool IsPatched { get; private set; } = false;
+
+        /// <summary>
+        /// Initializes a new <see cref="Spawnable"/>, the basic class needed for any item that can be spawned into the Subnautica game world.
+        /// </summary>
+        /// <param name="classId">The main internal identifier for this item. Your item's <see cref="TechType"/> will be created using this name.</param>
+        /// <param name="friendlyName">The name displayed in-game for this item whether in the open world or in the inventory.</param>
+        /// <param name="description">The description for this item; Typically seen in the PDA, inventory, or crafting screens.</param>
+        protected Spawnable(string classId, string friendlyName, string description)
+            : base(classId, $"{classId}Prefab")
+        {
+            if (string.IsNullOrEmpty(classId))
+            {
+                Logger.Error($"ClassID for Spawnables must be a non-empty value.");
+                throw new Exception($"Error patching Spawnable");
+            }
+
+            this.FriendlyName = friendlyName;
+            this.Description = description;
+
+            CorePatchEvents += RegisterBasics;
+        }
+
+        /// <summary>
+        /// This event triggers <c>before</c> the core patching methods begins.
+        /// You can attach simple <seealso cref="PatchEvent"/> methods to this event if you want to run code <c>before</c> the any of the core patching methods begin.
+        /// </summary>
+        protected PatchEvent OnStartedPatching;
+
+        /// <summary>
+        /// The main patching methods are executed here.
+        /// This event should only be used by the SMLHelper QuickStart classes.
+        /// </summary>
+        internal PatchEvent CorePatchEvents;
+
+        /// <summary>
+        /// This event triggers <c>after</c> the core patching methods begins.
+        /// You can attach simple <seealso cref="PatchEvent"/> methods to this event if you want to run code <c>after</c> the core patching methods have finished.
+        /// </summary>
+        protected PatchEvent OnFinishedPatching;
+
+        /// <summary>
+        /// Starts all patching code in SMLHelper.
+        /// If <seealso cref="IsPatched"/> is <c>true</c> then this method is skipped to avoid duplicate patching.
+        /// </summary>
+        /// <seealso cref="OnStartedPatching"/>
+        /// <seealso cref="OnFinishedPatching"/>
+        public void Patch()
+        {
+            if (this.IsPatched)
+                return; // Already patched. Skip.
+
+            OnStartedPatching?.Invoke();
+
+            // Because invocation order isn't guaranteed by event handlers,
+            // we make sure the TechType is patched first before anything else that might require it.
+            this.TechType = TechTypeHandler.Singleton.AddTechType(ModName, this.ClassID, this.FriendlyName, this.Description, false);
```

Comment:
This might be something I end up handling.  
Instead of this line here directly adding the TechType, instead it should be put into an `internal virtual` method `PatchTechType` that is overridden by `PDAItem`.  
Within the scope of `PDAItem`, we can pass in the correct "UnlocksAtStart" value.  
This current implementation is fine within `Spawnable` since use of this class alone isn't intended for items that are unlocked as blueprints.